presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -412,6 +449,12 @@ private OperationTimer createTimer()
                         throwIfUnchecked(throwable);
                         throw new RuntimeException(throwable);
                     }
+
+                    if (fragmentResultCachingEnabled) {
```

Comment:
Since this cache is on the driver's side, is there any sanity check in protection from oom? just curious if this cache size can be stored for monitoring so caching on the driver is in a controlled manner.

Reply:
Good point. `FragmentResultCacheManager` would be a singleton on each Presto worker, and cache instrumentation would be  part of the `FragmentResultCacheManager` implementation (which is still work-in-progress). 

Currently the plan is to store the map keys (plan-split pair) in memory and actual result pages in local disk. We should be able to set configuration to limit the cache entries so that we could keep memory usage in check.